packages/web/src/services/audius-backend/WithdrawUSDC.ts

```diff
@@ -0,0 +1,224 @@
+  )
+  return closeSolanaUSDCAccountInstr
+}
+
+// Swaps USDC for SOL, taking USDC from the user's USDC user bank and
+// depositing SOL into the user's root solana account.
+const swapUSDCForSol = async (
+  amount: number,
+  solanaRootAccountPubkey: PublicKey,
+  usdcUserBank: PublicKey,
+  solanaUSDCATA: PublicKey,
+  feePayerPubkey: PublicKey
+) => {
+  const connection = await getSolanaConnection()
+
+  // TODO: double check slippage
+  const quoteRoute = await JupiterSingleton.getQuote({
+    inputTokenSymbol: 'USDC',
+    outputTokenSymbol: 'SOL',
+    inputAmount: amount,
+    slippage: USDC_SLIPPAGE,
+    swapMode: SwapMode.ExactOut,
+    onlyDirectRoutes: true
+  })
+  const usdcNeededAmount = quoteRoute.inputAmount
+
+  const swapRoute = await JupiterSingleton.getQuote({
+    inputTokenSymbol: 'USDC',
+    outputTokenSymbol: 'SOL',
+    inputAmount: usdcNeededAmount.uiAmount,
+    slippage: USDC_SLIPPAGE,
+    onlyDirectRoutes: true
+  })
+  const exchangeInfo = await JupiterSingleton.exchange({
+    routeInfo: swapRoute.route,
+    userPublicKey: solanaRootAccountPubkey,
+    feeAccount: feePayerPubkey
+  })
+  const swapInstructions =
+    exchangeInfo.transactions.swapTransaction.instructions
+
+  const transferInstructions =
+    await libs().solanaWeb3Manager!.createTransferInstructionsFromCurrentUser({
+      amount: usdcNeededAmount.uiAmount,
+      feePayerKey: feePayerPubkey,
+      senderSolanaAddress: usdcUserBank,
+      recipientSolanaAddress: solanaUSDCATA,
+      instructionIndex: 1,
+      claimableTokenPDA: libs().solanaWeb3Manager!.claimableTokenPDAs.usdc,
+      solanaTokenProgramKey: TOKEN_PROGRAM_ID,
+      claimableTokenProgramKey:
+        libs().solanaWeb3Manager!.claimableTokenProgramKey,
+      connection,
+      mint: 'usdc'
+    })
+
+  const createATAInstruction = await createUSDCATAInstruction(
+    solanaUSDCATA,
+    solanaRootAccountPubkey,
+    feePayerPubkey
+  )
+  const closeATAInstruction = await closeUSDCATAInstruction(
+    solanaUSDCATA,
+    solanaRootAccountPubkey,
+    feePayerPubkey
+  )
+
+  const instructions = [
+    createATAInstruction,
+    ...transferInstructions,
+    ...swapInstructions,
+    closeATAInstruction
+  ]
+  return await libs().solanaWeb3Manager!.transactionHandler.handleTransaction({
+    instructions,
+    skipPreflight: true,
+    feePayerOverride: feePayerPubkey
+  })
+}
+
+const getUSDCATA = async (solanaRootAccountPubkey: PublicKey) => {
+  return await Token.getAssociatedTokenAddress(
+    ASSOCIATED_TOKEN_PROGRAM_ID,
+    TOKEN_PROGRAM_ID,
+    libs().solanaWeb3Manager!.mints.usdc,
+    solanaRootAccountPubkey
+  )
+}
+
+const createAndFundDestinationATA = async (
+  destinationPubkey: PublicKey,
+  feePayerPubkey: PublicKey
+) => {
+  // TODO: factor in existing sol balance
+  // TODO: might have to pay rent for root sol account, see BuyAudio.ts
+  const desiredSolAmount =
+    (await getRentAndTransferTransactionFee(destinationPubkey)) /
```

Comment:
nit: I think calling/reusing `getRent` and `getTransactionFee` and summing them here makes more sense than wrapping in another fn